### PR TITLE
feature/teachers-list

### PR DESCRIPTION
**Pull request by nadiakohli**

### CODE REVIEW

File: front-end/src/components/Teacher/index.jsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        </div>
+      </div>
+      <div className={styles.description}>
+        <h1>{fullName}</h1>
+        <span>{activity}</span>
+      </div>
+      <div className={styles.test1}>
+        <div className={styles.blockWrap}>
+          <img src={speechBubble} />
+          <span>533 Reviews</span>
```
Comment (andriydzhrvts) [inline front-end/src/components/Teacher/index.jsx:55]: We need to add all mock teacher data in teacher collection and fetch  from back-end

File: front-end/src/pages/TeachersList/index.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+// Components
+import Card from 'components/TeachersList/Card';
+
+// Styles
+import styles from './TeachersList.module.scss';
+
+const TeachersList = () => {
+  const [teachers, setTeachers] = useState([]);
+  console.log(teachers);
```
Comment (andriydzhrvts) [inline front-end/src/pages/TeachersList/index.jsx]: Console.log

File: front-end/src/services/teacherService.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import API, { API_URL } from 'API';
+
+export const teacherService = {
+  getAllTeacher: async () => {
+    try {
```
Comment (andriydzhrvts) [inline front-end/src/services/teacherService.js]: You don't need try catch block here

File: front-end/src/services/teacherService.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      const { data } = await API.get(`${API_URL}/api/teachers`);
+      return data;
+    } catch (e) {
+      console.log(e);
+    }
+  },
+  getTeacherById: async (id) => {
+    try {
```
Comment (andriydzhrvts) [inline front-end/src/services/teacherService.js]: And here

File: front-end/src/pages/TeachersList/index.jsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  };
+
+  useEffect(() => {
+    fetchTeachers();
+  }, []);
+
+  return (
+    <div className={styles.wrapper}>
+      <h1>Teachers</h1>
+      <div className={styles.cardWrap}>
```
Comment (andriydzhrvts) [inline front-end/src/pages/TeachersList/index.jsx]: Add loading

File: front-end/src/components/Teacher/Courses/Courses.module.scss [hunk 6]
```diff
@@ -0,0 +1,11 @@
+@import 'styles/font.scss';
```
Comment (andriydzhrvts) [inline front-end/src/components/Teacher/Courses/Courses.module.scss:1]: ^ All picture should be hosted on cloud

File: front-end/src/pages/TeachersList/index.jsx (hunk 2 above)
Comment (nadiakohli) [reply front-end/src/pages/TeachersList/index.jsx]: fix it

File: front-end/src/services/teacherService.js (hunk 3 above)
Comment (nadiakohli) [reply front-end/src/services/teacherService.js]: fix it

File: front-end/src/pages/TeachersList/index.jsx (hunk 5 above)
Comment (nadiakohli) [reply front-end/src/pages/TeachersList/index.jsx]: fix it

File: front-end/src/components/Teacher/Courses/Courses.module.scss (hunk 6 above)
Comment (nadiakohli) [reply front-end/src/components/Teacher/Courses/Courses.module.scss:1]: How to do it?

Comment (andriydzhrvts) [reply front-end/src/components/Teacher/Courses/Courses.module.scss:1]: You could ask in Pasha

File: front-end/src/services/teacherService.js (hunk 4 above)
Comment (nadiakohli) [reply front-end/src/services/teacherService.js]: fix it

File: front-end/src/components/Teacher/Courses/Courses.module.scss (hunk 6 above)
Comment (nadiakohli) [reply front-end/src/components/Teacher/Courses/Courses.module.scss:1]: fix it

File: front-end/src/components/Teacher/index.jsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+          <span>Report This Author</span>
+        </div>
+      </div>
+      <div className={styles.description}>
+        <h1>{fullName}</h1>
+        <span>{activity}</span>
+      </div>
+      <div className={styles.test1}>
+        <div className={styles.blockWrap}>
+          <img src='https://storage.googleapis.com/teachers-platform-40cbe.appspot.com/e5b18e70845a38cb3232b3625f2f40f9.png' />
```
Comment (andriydzhrvts) [inline front-end/src/components/Teacher/index.jsx]: Please create constants for photos which were upload on cloud and then use as variable in code

Comment (nadiakohli) [reply front-end/src/components/Teacher/index.jsx]: fix it

File: front-end/src/pages/TeachersList/index.jsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  return (
+    <div className={styles.wrapper}>
+      <h1>Teachers</h1>
+      {teachers.length ? (
+        <div className={styles.cardWrap}>
+          {teachers.length
+            ? teachers.map((teacher) => (
+                <Card
+                  key={teacher?._id}
+                  link={`/app/teachers/${teacher?._id}/overview`}
```
Comment (andriydzhrvts) [inline front-end/src/pages/TeachersList/index.jsx]: You could pass teacher id and create link in Card component

Comment (nadiakohli) [reply front-end/src/pages/TeachersList/index.jsx]: fix it

**Review by andriydzhrvts — APPROVED** [review]
Good work!

File: front-end/src/components/Teacher/index.jsx (hunk 7 above)
Comment (nadiakohli) [reply front-end/src/components/Teacher/index.jsx]: fix it

**Pull request merged**